test_runner/batch_others/test_wal_acceptor.py

```diff
@@ -380,8 +386,22 @@ def test_wal_removal(zenith_env_builder: ZenithEnvBuilder):
     ]
     assert all(os.path.exists(p) for p in first_segments)
 
-    http_cli = env.safekeepers[0].http_client()
+    if not auth_enabled:
+        http_cli = env.safekeepers[0].http_client()
+    else:
+        http_cli = env.safekeepers[0].http_client(
+            auth_token=env.auth_keys.generate_tenant_token(tenant_id))
+        http_cli_other = env.safekeepers[0].http_client(
+            auth_token=env.auth_keys.generate_tenant_token(uuid4().hex))
+        http_cli_noauth = env.safekeepers[0].http_client()
+
     # Pretend WAL is offloaded to s3.
+    if auth_enabled:
+        for cli in [http_cli_other, http_cli_noauth]:
+            with pytest.raises(cli.HTTPError, match='Forbidden|Unauthorized'):
+                cli.record_safekeeper_info(tenant_id,
+                                           timeline_id, {'backup_lsn': 'FFFFFFFF/FEFFFFFF'})
+        # TODO: ensure the record_safekeeper_info was not executed
```

Comment:
Can we do it somehow?

Reply:
TODO: yes, we can, `record_safekeeper_info` is mostly for manual testing/fixes, it just changes some variables inside the Safekeeper. So we should be able to read them before/after the change.